sample_qc/joint_sample_qc.py

```diff
@@ -365,47 +343,51 @@ def main(args):
+                                        **joint_ht.filter(joint_ht.data_type == 'genomes').key_by('s')[genome_ht.s])
 
     exome_sample_qc_ht = hl.read_table(qc_temp_data_prefix('exomes') + '.sample_qc.ht')
     genome_sample_qc_ht = hl.read_table(qc_temp_data_prefix('genomes') + '.sample_qc.ht')
 
-    exome_mt = exome_mt.annotate_cols(**exome_sample_qc_ht[exome_mt.s])
-    genome_mt = genome_mt.annotate_cols(**genome_sample_qc_ht[genome_mt.s])
-    exome_mt = exome_mt.annotate_cols(idx=exome_mt.qc_pop + "_" + hl.str(exome_mt.qc_platform))
-    genome_mt = genome_mt.annotate_cols(idx=genome_mt.qc_pop + "_" + hl.str(genome_mt.qc_platform))
+    exome_ht = exome_ht.annotate(**exome_sample_qc_ht[exome_ht.s])
+    genome_ht = genome_ht.annotate(**genome_sample_qc_ht[genome_ht.s])
 
     # For each population, aggregate sample QC metrics and calculate the MAD/mean/stdev
     logger.info('Calculating platform- and population-specific sample QC thresholds...')
     exome_qc_metrics = ['n_snp', 'r_ti_tv', 'r_insertion_deletion', 'n_insertion', 'n_deletion', 'r_het_hom_var']
-    exome_ht = calculate_metrics(exome_mt, exome_qc_metrics, 'exomes')
-
-    genome_qc_metrics = exome_qc_metrics + ['call_rate']
-    genome_ht = calculate_metrics(genome_mt, genome_qc_metrics, 'genomes')
+    exome_pop_platform_filter_ht = compute_stratified_metrics_filter(exome_ht, exome_qc_metrics, ['qc_pop', 'qc_platform'])
+    exome_ht = exome_ht.annotate_globals(hl.eval(exome_pop_platform_filter_ht.globals))
+    exome_ht = exome_ht.annotate(
+        **exome_pop_platform_filter_ht[exome_ht.key]
+    ).persist()
+
+    genome_qc_metrics = ['n_snp', 'r_ti_tv', 'r_insertion_deletion', 'n_insertion', 'n_deletion', 'r_het_hom_var']
+    genome_pop_platform_filter_ht = compute_stratified_metrics_filter(genome_ht, genome_qc_metrics, ['qc_pop', 'qc_platform'])
+    genome_ht = genome_ht.annotate_globals(hl.eval(genome_pop_platform_filter_ht.globals))
+    genome_ht = genome_ht.annotate(
+        **genome_pop_platform_filter_ht[genome_ht.key]
+    ).persist()
 
     # Annotate samples that fail their respective filters
-    logger.info('Flagging samples failing pop/platform-specific sample qc thresholds...')
-    exome_mt = apply_pop_platform_filters(exome_ht, exome_mt, exome_qc_metrics)
-    checkpoint = exome_mt.aggregate_cols(hl.agg.count_where(hl.len(exome_mt.pop_platform_filters) == 0))
-    logger.info('{0} exome samples found passing pop/platform-specific filtering'.format(checkpoint))
-    exome_mt.cols().annotate(data_type='exomes').key_by('data_type', 's').write(qc_ht_path('exomes', 'pop_platform'), args.overwrite)
+    checkpoint = exome_ht.aggregate(hl.agg.count_where(hl.len(exome_ht.pop_platform_filters) == 0))
+    logger.info(f'{checkpoint} exome samples found passing pop/platform-specific filtering')
+    exome_ht.annotate(data_type='exomes').key_by('data_type', 's').write(qc_ht_path('exomes', 'pop_platform'), args.overwrite)
```

Comment:
I like `exome_ht.key_by(data_type='exomes', s=exome_ht.s).write(...)` but that might just be me